site/src/Librecores/ProjectRepoBundle/RepoCrawler/RepoCrawler.php

```diff
@@ -89,4 +98,14 @@ public function updateSourceRepo()
     {
         // the default implementation is empty
     }
+
+    /**
+     * Access ProjectMetricsProvider class in GitRepoCrawler and GithubRepoCrawler
+     *
+     * @return ProjectMetricsProvider
+     */
+    public function getProjectMetricsProvider()
+    {
+        return $this->projectMetricsProvider;
+    }
```

Comment:
Why is that function needed?  Within the subclasses you can access the metrics provider through `$this->projectMetricsProvider` directly.